client/pull.go

```diff
@@ -103,20 +103,24 @@ type Downloader struct {
 }
 
 // httpGetRangeRequest performs HTTP GET range request to URL specified by 'u' in range start-end.
-func httpGetRangeRequest(ctx context.Context, url string, start, end int64) (*http.Response, error) {
+func (c *Client) httpGetRangeRequest(ctx context.Context, url string, start, end int64) (*http.Response, error) {
 	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
 	if err != nil {
 		return nil, err
 	}
 
 	req.Header.Add("Range", fmt.Sprintf("bytes=%d-%d", start, end))
 
-	return http.DefaultClient.Do(req)
+	if c.AuthToken != "" {
+		req.Header.Add("Authorization", fmt.Sprintf("BEARER %v", c.AuthToken))
+	}
+
+	return c.HTTPClient.Do(req)
```

Comment:
We should probably use `(*Client).newRequest` rather than `http.NewRequestWithContext` above, which will set the `Authorization` and also the `User-Agent`.

With that, I believe this can be simplified to:

```suggestion
	return c.HTTPClient.Do(req.WithContext(ctx))
```

Reply:
I hope I didn't overdo it :)